astroquery/gaia/core.py

```diff
@@ -357,7 +369,9 @@ def get_datalinks(self, ids, verbose=False):
         return self.__gaiadata.get_datalinks(ids=ids, verbose=verbose)
 
     def __query_object(self, coordinate, radius=None, width=None, height=None,
-                       async_job=False, verbose=False, columns=[]):
+                       async_job=False, verbose=False, columns=[],
+                       table_name=None, ra_column_name=None,
```

Comment:
we don't do this with the other modules and I don't think it's the overall preference to bloat up the number of kwargs with configurable members. cc @keflavich

Reply:
I agree. I see the motivation to have a configurable default table and row limit (:+1: for those), but not for the RA/Dec column name.  If there's a specific reason to make that configurable (i.e., if the RA/Dec columns have version numbers in them), then that would be a good enough reason, though.

Reply:
the main logic was to just use another instance for that, right? (not that `gaia_xyz = Gaia()` works, it doesn't, unlike other modules). If useful, we could even make it into the ``__init__`` as we do it in e.g. `Vizier`, to make it very convenient.

Reply:
The only reason I added the `*_column_name` arguments to the `query_object` functions was that the `cone_search` functions already had them, and having a common set of arguments for both seemed like a good idea.

That being said I don't think those config options should exist at all. It seems to me that having the coordinate system as a config option and/or a keyword argument would be beneficial, but the names of the longitude and latitude coordinates should be automatically taken from some built-in dictionary instead of having to rely on the user to supply both. If this is an idea we wish to pursue further in some other pull request then we would have to deprecate the `*_column_name` config options, and adding more public keyword arguments in this pull request that would have to be deprecated in the future would be counterproductive.

The best thing to do here seems to be to undo the addition of the `*_column_name` arguments to the `query_object` functions.

Reply:
I agree with the idea that coordinate system selection should be by system, not pairs of columns.

regarding configuration of the default table / row limit: those are, in many packages, accessible as _both_ configuration items and per-instance items.  The table selection is more like a base URL choice, which is always(?) configurable at the config level.